lib/lambda_deployment/configuration.rb

```diff
@@ -15,7 +15,7 @@ def load_config(config_file)
     end
 
     def alias_name
-      ENV['TAG'].to_s[/\d.*/]
+      ENV['TAG'].to_s.gsub(/[^\da-zA-Z\-_]/, '')[/([0-9]+)([a-zA-Z0-9\-_]+)/]
```

Comment:
add a comment on why this crazy regex is there ... and maybe a test for the edge-cases we discussed